File: CSharp/Console Minesweeper/Tile.cs
using System;

public class Tile
{
    protected char fieldValue = ' ';
    protected int tileNum = 0;
    protected bool bombHere = false;
    protected bool hidden = true;
    protected bool flagged = false;

    public char FieldValue
    {
        get
        {
            if (Hidden == true) fieldValue = ' ';
            if (Hidden == false)
                if (BombHere == true) fieldValue = 'X';
                else fieldValue = char.Parse(TileNum.ToString());
            return fieldValue;
        }
    }

    public int TileNum
    {
        get
        {
            return tileNum;
        }
        set
        {
            tileNum = value;
        }
    }

    public bool BombHere
    {
        get
        {
            return bombHere;
        }
        set
        {
            bombHere = value;
        }
    }

    public bool Hidden
    {
        get
        {
            return hidden;
        }
    }

    public bool Flagged
    {
        get
        {
            return flagged;
        }
    }

    public void Reveal()
    {
        hidden = false;
    }

    public void Hide()
    {
        hidden = true;
    }

    public void Reset()
    {
        TileNum = 0;
        BombHere = false;
        Hide();
    }

    public void Flag()
    {
           
    }
}

File: Java/Tile.java
public class Tile
{
    public final int Number;
    public final boolean Mined;
    private boolean revealed;
    private boolean flagged;
    
    public Tile(boolean mined, int number)
    {
        Mined = mined;
        Number = number;
    }

    public boolean Hidden()
    {
        return !revealed;
    }

    public boolean Flagged()
    {
        return flagged;
    }

    public void Reveal()
    {
        if (!flagged) revealed = true;
    }

    public void Hide()
    {
        revealed = false;
    }

    public void Flag()
    {
        if (!revealed) flagged = true;
    }

    public void Unflag()
    {
        flagged = false;
    }
}